pandas/tests/io/excel/test_readers.py

```diff
@@ -1556,6 +1556,12 @@ def test_excel_read_binary_via_read_excel(self, read_ext, engine):
         expected = pd.read_excel("test1" + read_ext, engine=engine)
         tm.assert_frame_equal(result, expected)
 
+    def test_read_excel_header_index_out_of_range(self, engine):
+        # GH#43143
+        with open("df_header_oob.xlsx", "rb") as f:
```

Comment:
Is there a benefit of adding a new excel file here? In other words, could we use already existing xlsx in data/excel and specify a header like `list(range(100))`?

Reply:
In theory yes, but I'd like to avoid dependencies between the tests. In my experience, that gets you into trouble at some point